PrettyPrint/pretty_print.py

```diff
@@ -0,0 +1,80 @@
+"""
+This module is for printing texts in Markdown or HTML.
+"""
+
+import markdown
+
+__author__ = ['Andrew Deloucas <ADeloucas@g.harvard.com>']
+__license__ = 'MIT License. See LICENSE.'
+
+
+class PrettyPrint(object):
+    """
+    Prints texts in markdown or in HTML.
+    """
+    def __init__(self):
+        """
+        Empty.
+        """
+
+    def markdown_print_file(self, ingest_text, filename):
+        """
+        Prints whole text_file in markdown.
+        :param ingest_text: text ingested by cdli_corpus
+        :param filename: where you wish to save the markdown data
+        :return: output in filename.md
+       """
+        with open(filename, mode='r+', encoding='utf8') as text_file:
+            for text in ingest_text:
+                edition = text['text edition'][0]
+                metadata = '\n \t'.join(text['metadata'][0])
+                transliteration = '\n \t'.join(text['transliteration'][0])
+                m_d = """
+{edition}
+---
+###### metadata
+    {metadata}
+###### transliteration
+    {transliteration} 
+""".format(edition=edition, metadata=metadata, transliteration=transliteration)
+                self.markdown_file = m_d
+                text_file.write(self.markdown_file)
+
+    def markdown_print_single_text(self, ingest_text, cdli_number, filename):
+        """
+        Prints single text in file in markdown.
+       :param ingest_text: text ingested by cdli_corpus
+       :param cdli_number: text you wish to print
+       :param filename: where you wish to save the markdown data
+       :return: output in filename.md
+       """
+        with open(filename, mode='r+', encoding='utf8') as text_file:
+            for text in ingest_text:
+                cdli = text['cdli number'][0]
+                if cdli_number in cdli:
+                    edition = text['text edition'][0]
+                    metadata = '\n \t'.join(text['metadata'][0])
+                    transliteration = '\n \t'.join(text['transliteration'][0])
+                    m_d = """
+{edition}
+---
+###### metadata
+    {metadata}
+###### transliteration
+    {transliteration} 
+""".format(edition=edition, metadata=metadata, transliteration=transliteration)
+                    self.markdown_text = m_d
+                    text_file.write(self.markdown_text)
+
+    def html_print(self, origin, destination):
```

Comment:
Can you include a method that converts the text directly to html without saving it as a markdown file?

Reply:
I'm going off of markdown.markdown; is it wiser to simply custom this method?

Reply:
All you need to do is just go from the markdown string created by the lib to html rather than save to a markdown file.

Reply:
Understood -- when we are satisfied with the html output, I'll remove the markdown file aspects of the markdown classes, leaving the final output as `self.markdown_text` and `self.markdown_file`.

Reply:
Yeah I think that sounds good. The class should be able to produce the markdown and html as strings as well as save the output to some local file. You'll have to decide on the best way to do that.

Here's a quick sketch of what I would do:
```python
class PrettyPrint:
    def __init__(self, text)

    def to_markdown(save_output=False):
        # returns md of text as str, optionally saves to disk
    def to_html(save_output=False):
        # returns html of text as str, optionally saves to disk

# Example usage
pp = PrettyPrint(text='some text')
md = pp.to_markdown() # str of md text
pp.to_html(save_output=True) # created some_text.html with html text
```